### PR TITLE
Np 35376 cristin import missing field created date published date

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java [hunk 1]
```diff
@@ -102,16 +105,35 @@ private Organization extractOrganization() {
...
+                   .map(this::localDateToInstant)
+                   .orElseGet(() -> convertPublishedYearInstant(cristinObject));
+    }
+
+    private Instant convertPublishedYearInstant(CristinObject cristinObject) {
+
+        return Optional.ofNullable(cristinObject.getPublicationYear())
+                   .map(year -> LocalDate.of(year, Month.JANUARY, FIRST_DAY_OF_MONTH)
+                                    .atStartOfDay()
+                                    .toInstant(zoneOffset()))
```
Comment (truhacevkir) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java:125]: Kanskje lage metode toInstant eller lignende?

                   .map(this::toInstant)
                   .orElse(null);
    }

    private Instant toInstant(Integer year) {
        return LocalDate.of(year, Month.JANUARY, FIRST_DAY_OF_MONTH)
                   .atStartOfDay()
                   .toInstant(zoneOffset());
    }

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java [hunk 2]
```diff
@@ -102,16 +105,35 @@ private Organization extractOrganization() {
...
         return Optional.ofNullable(cristinObject.getEntryCreationDate())
-                   .map(ld -> ld.atStartOfDay().toInstant(zoneOffset()))
+                   .map(this::localDateToInstant)
+                   .orElseGet(() -> extractPublishedDateOrConvertYearToInstant(cristinObject));
+    }
+
+    private Instant extractPublishedDateOrConvertYearToInstant(CristinObject cristinObject) {
+        return Optional.ofNullable(cristinObject.getEntryPublishedDate())
+                   .map(this::localDateToInstant)
+                   .orElseGet(() -> convertPublishedYearInstant(cristinObject));
```
Comment (truhacevkir) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java:117]: Kan ikke denne metoden hete bare extractPublishedDate

**Review by truhacevkir — APPROVED** [review]

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java (hunk 2 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java]: Fikset

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java (hunk 1 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java]: Fikset

**Pull request merged**